web/client/src/components/form/TokenForm.jsx

```diff
@@ -30,8 +36,35 @@ export default function TokenForm() {
         tokenAudience: "",
         tokenCredentials: "",
       }}
-      onSubmit={async () => {
-        await sleep(3000);
+      onSubmit={async (values) => {
+        const tokenCred = JSON.parse(values.tokenCredentials);
+        let finalCredentials;
+        if (tokenCred["web"] !== undefined) {
```

Comment:
Can we turn this into a helper function and add a comment? the credentials can be nested under other types besides "web" right? For example "installed"?

Reply:
Done!